### PR TITLE
Make common macros extendable

### PR DESCRIPTION
**Pull request by amosbird**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Enable the ability to extend profile events, metrics, error codes, etc. from external sources outside of the ClickHouse codebase.

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 3eef78ae185a8d3e5762f24af59ee08dc835c434 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51646/3eef78ae185a8d3e5762f24af59ee08dc835c434/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Common/StatusInfo.cpp [hunk 1]
```diff
@@ -2,9 +2,14 @@
 #include <Common/ExternalLoaderStatus.h>
 
 /// Available status. Add something here as you wish.
-#define APPLY_FOR_STATUS(M) \
+#define APPLY_FOR_BUILTIN_STATUS(M) \
     M(DictionaryStatus, "Dictionary Status.", DB::getStatusEnumAllPossibleValues()) \
 
+#ifdef APPLY_FOR_EXTERNAL_STATUS
```
Comment (alexey-milovidov) [inline src/Common/StatusInfo.cpp:8]: I see StatusInfo as unneeded - it was added for only one unimportant use-case; the code is not good (mostly copy-paste), and I'd like to remove it eventually.

Comment (amosbird) [reply src/Common/StatusInfo.cpp]: OK. I've reverted the changes in StatusInfo.

**Comment by amosbird** [comment]
Test failures are unlikely related to this PR. Could you please take another look @alexey-milovidov ?

**Comment by alexey-milovidov** [comment]
LGTM, but we have to fix all the tests before merging...

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**